faunadb-java-dsl/src/main/java/com/faunadb/client/types/Codec.java

```diff
@@ -167,4 +229,172 @@ public R apply(T input) {
     };
   }
 
+  static <I extends O, O> Function<I, O> cast() {
+    return (Function<I, O>) Functions.identity();
```

Comment:
This gives me a warning because it brakes the contract of `identity`. I'd either implement with `new Function...` of add a `@SuppressWarnings("unchecked")` to the method.